config/samples/bgp_full_sample_config.yaml

```diff
@@ -0,0 +1,54 @@
+
+---
+apiVersion: metallb.io/v1alpha1
+kind: AddressPool
+metadata:
+  name: addresspool-sample1
+  namespace: metallb-system
+spec:
+  protocol: bgp
+  addresses:
+    - 172.18.0.100-172.18.0.255
+  bgpAdvertisements:
+  - communities:
+    - acceptOAM
+    - noAdvertise
+---
+apiVersion: metallb.io/v1alpha1
+kind: BGPPeer
+metadata:
+  name: peer-sample1
+  namespace: metallb-system
+spec:
+  peerAddress: 10.0.0.1
+  peerASN: 64501
+  myASN: 64500
+  routerID: 10.10.10.10
+
+---
+apiVersion: metallb.io/v1alpha1
+kind: BGPPeer
+metadata:
+  name: peer-sample2
+  namespace: metallb-system
+spec:
+  peerAddress: 11.0.0.1
+  peerASN: 64502
+  myASN: 64000
+  routerID: 20.20.20.20
+
+---
+apiVersion: metallb.io/v1alpha1
+kind: BGPCommunity
+metadata:
+  name: bgpcommunity-sample1
+spec:
+  acceptOAM: 7003:007
```

Comment:
I think we can avoid `BGPCommunity`.  The value it provides in the configmap is giving a more meaningful name to something that's otherwise just two numbers `a:b`.

We can allow meaningful names when configuring communities in BGP advertisements without a new CRD to define them by having a `name` and `value` field (or some better name).

```
  communities:
    - value: 64512:1
    - name: no-export
      value: 65535:65281
```

What you have would work fine.  It just feels like overkill to make it a separate CRD to me.

Reply:
to make sure I understand you suggesting addresspool CR config will be ?
```
bgpAdvertisements:
  - communities:
    - name: no-export
      value: 65535:65281
    - name: accept-oam
      value: 7003:007
```

Reply:
Yes, but you could also make `name` optional.

Reply:
here is my understanding from configmap example, the name in address-pool is how u tie it to bgp-communities settings so we need both name and value correct ?

```
address-pools:
    - name: oam
      protocol: bgp
      auto-assign: false
      addresses:
      - 10.137.48.243-10.137.48.255
      bgp-advertisements:
      - communities:
        - accept-oam

bgp-communities:
      accept-oam: 7003:007
```